### DISCUSSION TITLE
Query on OCSP signer certificate

### DISCUSSION THREAD

**Discussion by nknkgithub** [strongswan/strongswan#2209] [General]
We have enabled ocsp_check in strongswan.conf and " revocation = strict" is set in swanctl.conf.
strongswan.conf

We have uploaded "Root certificate" of the OCSP response signer certificate in the x509ca folder and x509ocsp folder.
```
swanctl --list-certs
...
List of X.509 CA Certificates
subject:  "C=US, O=<O>, CN=<Root Cert>"
issuer:   "C=US, O=<O>, CN=<Root Cert>"
...  
```

Below error is observed during tunnel establishment
`ocsp response verification failed, no signer certificate 'C=US, O=<O>, CN=<Signer CN>' found`

We dont have signer certificate beforehand and it is included in the OCSP response.

Can you please clarify why OCSP response verification is failing even though the root certificate of the signer is loaded in the x509ca folder?

**Comment by tobiasbrunner** [comment]
OCSP responses have to be signed by either the CA certificate directly, by a OCSP signer certificate (with OCSPSigning extended key usage flag) that is issued by the CA, or by a self-signed OCSP signer certificate. They can't be signed by a certificate from a completely unrelated CA.

**Comment by nknkgithub** [comment]
Thanks @tobiasbrunner .

We have enabled ocsp_check in strongswan.conf and " revocation = **ifuri**" is set in swanctl.conf.


OCSP signer certificate(OCSP signer cert) is issued by subCA1.Root certificate of this subCA1 (i.e. rootCA) is uploaded in the x509ca folder and x509ocsp folder.

We observe OCSP verification is successful for device cert.
```
08[CFG] <home3|1> checking certificate status of "C=<C>, O=<O>, OU=<OU>, OU=<OU1>, CN=<peer device cert>"
08[CFG] <home3|1>   requesting ocsp status from '<OCSP URI>' ...
08[CFG] <home3|1>   ocsp response correctly signed by "C=<C>, O=<O>, OU=<OU>, OU=<OU1>, CN=<OCSP signer cert>"
08[CFG] <home3|1>   ocsp response is valid: until May 19 10:24:32 2024
08[CFG] <home3|1> certificate status is good
```
But below error is observed later
```
08[CFG] <home3|1>   using trusted ca certificate "C=<C>, O=<O>, OU=<OU>, OU=<OU1>, CN=<rootCA>"
08[CFG] <home3|1> checking certificate status of "C=<C>, O=<O>, OU=<OU>, OU=<OU1>, CN=<subCA1>"
08[CFG] <home3|1> ocsp response verification failed, no signer certificate 'C=<C>, O=<O>, OU=<OU>, OU=<OU1>, CN=<OCSP signer cert>' found
```

Tunnel is established.
```
[IKE] authentication of 'C=<C>, O=<O>, OU=<OU>, OU=<OU1>, CN=<peer device cert>' with RSA_EMSA_PKCS1_SHA2_256 successful
[IKE] IKE_SA home3[1] established between <device_IP>[<device_CN>]...10.197.111.19[C=<C>, O=<O>, OU=<OU>, OU=<OU1>, CN=<peer device cert>]
```

Though the tunnel is established, why this error is observed?

**Reply by tobiasbrunner** [reply]
Please see the commit message of 9e17a0ed88ee6c9c2c77fbb747e667a339819b49 (released with 5.9.11) that explains this when handling cached OCSP responses.
